src/subdomains/generic/user/models/user-data/user-data-notification.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@nestjs/common';
+import { Interval } from '@nestjs/schedule';
+import { UserDataRepository } from './user-data.repository';
+import { Lock } from 'src/shared/utils/lock';
+import { MailType } from 'src/subdomains/supporting/notification/enums';
+import { NotificationService } from 'src/subdomains/supporting/notification/services/notification.service';
+import { Config, Process } from 'src/config/config';
+
+@Injectable()
+export class UserDataNotificationService {
+  constructor(
+    private readonly userDataRepo: UserDataRepository,
+    private readonly notificationService: NotificationService,
+  ) {}
+
+  @Interval(300000)
+  @Lock(1800)
+  async sendNotificationMails(): Promise<void> {
+    if (Config.processDisabled(Process.BLACK_SQUAD_MAIL)) return;
+    await this.blackSquadInvitation();
+  }
+
+  private async blackSquadInvitation(): Promise<void> {
+    const entities = await this.userDataRepo
+      .createQueryBuilder('userData')
+      .select('userData')
+      .where('userData.blackSquadMailSendDate IS NULL')
+      .andWhere('userData.buyVolume + userData.sellVolume + userData.cryptoVolume >= :limit', {
+        limit: Config.support.blackSquad.limit,
+      })
```

Comment:
In UserService.getUserDetails, we only use buy and sell volume (without crypto) and have `>` instead of `>=`. Should be the same for both.